crates/solidity/outputs/cargo/crate/src/generated/language.rs

```diff
@@ -1573,29 +1577,55 @@ impl Language {
                 }),
             )
         };
-        let parse_exponentiation_operator_removed_from_0_6_0 = |input: &mut ParserContext| {
+        let parse_exponentiation_expression_removed_from_0_6_0 = |input: &mut ParserContext| {
             PrecedenceHelper::to_binary_operator(
                 RuleKind::BinaryExpression,
                 25u8,
                 25u8 + 1,
-                self.parse_token_with_trivia::<LexicalContextType::Default>(
-                    input,
-                    TokenKind::AsteriskAsterisk,
-                ),
-            )
-        };
-        let parse_exponentiation_operator_introduced_from_0_6_0 = |input: &mut ParserContext| {
-            PrecedenceHelper::to_binary_operator(
-                RuleKind::BinaryExpression,
-                27u8 + 1,
+                ChoiceHelper::run(input, |mut choice, input| {
+                    if !self.version_is_at_least_0_6_0 {
```

Comment:
I see we are checking `self.version_is_at_least_0_6_0` twice, both before and after calling `parse_exponentiation_expression_removed_from_0_6_0`. Is this intended? Shouldn't the outer check be enough in this case?

Reply:
Same thing in `parse_prefix_expression_removed_from_0_5_0`

Reply:
This stems from the difference in v1/v2 definitions. I'm bridging v2-to-v1 as using a single definition for the operator (as a choice over different versioned parsers) but including it multiple times, using the versioning mechanism from v1 (which generates the extra `_{removed,introduced}_*` sub-parsers), for the precedence expression.

It's correct albeit a bit redundant in places like this; once v1 is gone, I want to adapt the existing parser codegen grammar interface to better suit the v2 model and clean this up.

Reply:
Makes sense. Thanks for explaining!